src/modules/fancyzones/lib/ZoneWindowDrawing.cpp

```diff
@@ -218,53 +218,63 @@ void ZoneWindowDrawing::RenderLoop()
 void ZoneWindowDrawing::Hide()
 {
     _TRACER_;
-    std::unique_lock lock(m_mutex);
-
-    m_animation.reset();
-
-    if (m_shouldRender)
+    bool shouldHideWindow;
     {
+        std::unique_lock lock(m_mutex);
+        m_animation.reset();
+        shouldHideWindow = m_shouldRender;
         m_shouldRender = false;
+    }
+
+    if (shouldHideWindow)
+    {
         ShowWindow(m_window, SW_HIDE);
     }
 }
 
 void ZoneWindowDrawing::Show(unsigned animationMillis)
 {
     _TRACER_;
-    std::unique_lock lock(m_mutex);
-    
-    if (!m_shouldRender)
+    bool shouldShowWindow;
     {
-        ShowWindow(m_window, SW_SHOWNA);
-    }
+        std::unique_lock lock(m_mutex);
+        shouldShowWindow = !m_shouldRender;
+        m_shouldRender = true;
 
-    animationMillis = max(animationMillis, 1);
+        animationMillis = max(animationMillis, 1);
+        
+        if (!m_animation || !m_animation->fadeIn)
+        {
+            m_animation.emplace(AnimationInfo{ std::chrono::steady_clock().now(), animationMillis, true });
+        }
+    }
 
-    if (!m_animation || !m_animation->fadeIn)
+    if (shouldShowWindow)
     {
-        m_animation.emplace(AnimationInfo{ std::chrono::steady_clock().now(), animationMillis, true });
+        ShowWindow(m_window, SW_SHOWNA);
     }
 
-    m_shouldRender = true;
     m_cv.notify_all();
 }
 
 void ZoneWindowDrawing::Flash(unsigned animationMillis)
 {
     _TRACER_;
-    std::unique_lock lock(m_mutex);
+    bool shouldShowWindow;
```

Comment:
```suggestion
    bool shouldShowWindow = false;
```

Reply:
What about `bool shouldShowWindow{};`?
It's getting initialized by the code below either way, and the compiler knows this and is not complaining. Adding `{}` is probably the most discreet way to initialize it.